elf_reader.go

```diff
@@ -610,28 +657,70 @@ func mapSpecFromBTF(spec *btf.Spec, name string) (*MapSpec, error) {
 			}
 
 		case "pinning":
+			if inner {
+				return nil, errors.New("inner maps can't be pinned")
+			}
+
 			pinning, err := uintFromBTF(member.Type)
 			if err != nil {
 				return nil, fmt.Errorf("can't get pinning: %w", err)
 			}
 
 			pinType = PinType(pinning)
 
-		case "key", "value":
+		case "values":
```

Comment:
Is it valid to have both `values` and `value` or `value_size` specified?

Reply:
`value` is resolved into a `value_size`, so no issue there. libbpf enforces the following: `if (map->def.value_size && map->def.value_size != 4) {`. Will implement this check as well.

Reply:
Turns out I was also missing a `valueSize = 4` assignment as well.